src/mscorlib/shared/System/Text/ValueStringBuilder.cs

```diff
@@ -21,7 +22,25 @@ public ValueStringBuilder(Span<char> initialBuffer)
             _pos = 0;
         }
 
-        public int Length => _pos;
+        public int Length
+        {
+            get => _pos;
+            set
+            {
+                EnsureCapacity(value);
+                _pos = value;
+            }
```

Comment:
What does it mean to set the length like this? Couldn't this be filled with garbage?

Reply:
I would drop the `EnsureCapacity` here - require the space to be preallocated and filled in by other means.

Reply:
Will do.